src/test/scala/cognite/spark/v1/FilesRelationTest.scala

```diff
@@ -39,74 +55,300 @@ class FilesRelationTest extends FlatSpec with Matchers with SparkTest {
+             |select ${id.toString} as id,
+             |'text/plain-$source' as mimeType,
+             |'upserted-$source' as source
+     """.stripMargin)
+        .write
+        .format("cognite.spark.v1")
+        .option("apiKey", writeApiKey)
+        .option("type", "files")
+        .option("onconflict", "upsert")
+        .save()
+
+      //check updated data
+      val updatedWithUpsert =
+        retryWhile[Array[Row]](
+          spark
+            .sql(
+              s"select * from destinationFiles where mimeType = 'text/plain-$source' and source = 'upserted-$source'")
+            .collect,
+          df => df.length < 1)
+      assert(updatedWithUpsert.length == 1)
+
+      //original data doesn't exist
+      val updated = retryWhile[Array[Row]](
+        spark
+          .sql(s"select * from destinationFiles where source = '$source'")
+          .collect,
+        df => df.length > 0)
+      assert(updated.isEmpty)
+    } finally {
+      try {
+        cleanupFiles(s"upserted-$source")
+      } catch {
+        case NonFatal(_) => // ignore
+      }
+    }
 
   }
+
+  it should "support deletes" taggedAs WriteTest in {
+    val source = s"delete-${shortRandomString()}"
+
+    try {
+      spark
+        .sql(s"""
+                |select "name-$source" as name,
+                |null as id,
+                |'$source' as source,
+                |'externalId-$source' as externalId,
+                |null as mimeType,
+                |null as metadata,
+                |null as assetIds,
+                |null as datasetId,
+                |null as sourceCreatedTime,
+                |null as sourceModifiedTime,
+                |null as securityCategories,
+                |null as uploaded,
+                |null as createdTime,
+                |null as lastUpdatedTime,
+                |null as uploadedTime,
+                |null as uploadUrl
+     """.stripMargin)
+        .select(sourceDf.columns.map(col): _*)
+        .write
+        .insertInto("destinationFiles")
+
+      val rows = retryWhile[Array[Row]](
+        spark.sql(s"select * from destinationFiles where source = '$source'").collect,
+        rows => rows.length < 1)
+      assert(rows.length == 1)
+
+      //Delete using id
+      spark
+        .sql(s"""
+                |select id
+                |from destinationFiles
+                |where source = '$source'
+     """.stripMargin)
+        .write
+        .format("cognite.spark.v1")
+        .option("apiKey", writeApiKey)
+        .option("type", "files")
+        .option("onconflict", "delete")
+        .save()
+
+      val idsAfterDelete =
+        retryWhile[Array[Row]](
+          spark
+            .sql(s"select id from destinationFiles where source = '$source'")
+            .collect,
+          df => df.length > 0)
+      assert(idsAfterDelete.isEmpty)
+    } finally {
+      try {
+        cleanupFiles(source)
+      } catch {
+        case NonFatal(_) => // ignore
+      }
+    }
+  }
+
```

Comment:
Would it be possible/reasonable to have a test like https://github.com/cognitedata/cdp-spark-datasource/blob/master/src/test/scala/cognite/spark/v1/EventsRelationTest.scala#L1100 for Files schemas?

Reply:
It makes sense to check that all necessary parameters are there in each schema, but as the name is required parameter in files how should I go about it?